lib/ansible/plugins/loader.py

```diff
@@ -276,9 +324,9 @@ def _find_plugin(self, name, mod_type='', ignore_deprecated=False, check_aliases
                 full_name = os.path.basename(full_path)
 
                 # HACK: We have no way of executing python byte compiled files as ansible modules so specifically exclude them
-                # FIXME: I believe this is only correct for modules and module_utils.
                 # For all other plugins we want .pyc and .pyo should be valid
-                if full_path.endswith(('.pyc', '.pyo')):
+                # since module_utils and modules dont have a base class but rest of plugins do:
+                if not self.base_class and full_path.endswith(('.pyc', '.pyo')):
```

Comment:
if this is legit, we should probably have a test for it, if not, this seems like it could cause more problems than it solves...

Reply:
Yeah, this seems like one of those unrelated changes, that shouldn't necessarily go here.  I'll have to verify if it breaks become plugins by reverting this change for now.

Reply:
This change has been removed.